File: strongly-typed-typescript/exhaustiveness/unsafe_multiple.ts
enum Transport {
  AIR,
  GROUND,
}

enum State {
  ACTIVE,
  INACTIVE,
}

let transport: Transport = Math.random() < 0.5 ? Transport.AIR : Transport.GROUND;
let state: State = Math.random() < 0.5 ? State.ACTIVE : State.INACTIVE;
let message: String;
switch([transport, state]) {
  case [Transport.AIR, State.ACTIVE]:
    message = "🛩!";
    break;
  case [Transport.GROUND, State.ACTIVE]:
    message = "🚙!";
    break;
  case [Transport.AIR, State.INACTIVE]:
    message = "🛩";
    break;
  case [Transport.GROUND, State.INACTIVE]:
    message = "🚙";
    break;
}

console.log(message);

File: strongly-typed-kotlin/src/exhaustiveness/unsafe_multiple.kt
enum class Transport {
    Air, Ground;
}
enum class State {
    Active, Inactive;
}

fun main() {
    val transport: Transport = if (Math.random() < 0.5) Transport.Air else Transport.Ground
    val state: AccountStatus = if (Math.random() < 0.5) AccountStatus.Active else AccountStatus.Inactive
    val message: String = when(Pair(transport, state)) {
        Pair(Transport.Air, AccountStatus.Active) -> "🛩✅"
        Pair(Transport.Ground, AccountStatus.Active) -> "🚙✅"
        Pair(Transport.Air, AccountStatus.Inactive) -> "🛩❌"
        Pair(Transport.Ground, AccountStatus.Inactive) -> "🚙❌"
        else -> "🤷"
    }

    println(message)
}
